### PR TITLE
411 use remotes installs

### PR DESCRIPTION
**Pull request by Felixmil**
## Simpler package installation:

This is now enough to install `{esqlabsR}`
```r
install.packages("devtools")
install.packages('https://github.com/Open-Systems-Pharmacology/rClr/releases/download/v0.9.2/rClr_0.9.2.zip', repos = NULL, type = 'binary')
devtools::install_github("esqLABS/esqlabsR")
```

Test it with:
```r
install.packages("devtools")
install.packages('https://github.com/Open-Systems-Pharmacology/rClr/releases/download/v0.9.2/rClr_0.9.2.zip', repos = NULL, type = 'binary')
devtools::install_github("esqLABS/esqlabsR@7f676b5", dependencies = TRUE)

library(esqlabsR)
```

## Clean dependency graph
- Automatically installs HEAD (develop branch) of `{ospsuite.utils}`, `{tlf}`, `{ospsuite}` and `{parameter.identification}` github repos,
- All OpenPharma packages's dependencies are installed automatically,
- Showtext is now a suggest packages in order to make snaps from `{tlf}` pass tests.
- `{shiny}` and `{shinyjs}`  are now suggest dependencies because required only for the embeded shinyApps.
    ```r
    esqlabsR::startUnitConverter()
    ℹ The package "shinyjs" is required to launch this shiny App
    ✖ Would you like to install it?

    1: Yes
    2: No
    ```
 
## Mak all this work on AppVeyor
- Refactor `Rscript` calls into one `install_ext_deps.R` file,
- Added a GITHUB_PAT environment variable in AppVeyor project settings.

### CODE REVIEW

**Comment by Felixmil** [comment]
``` r
install.packages("devtools")
#> Installing package into 'C:/Users/Felix/Code/test_install_esqlabsR/renv/library/R-4.2/x86_64-w64-mingw32'
#> (as 'lib' is unspecified)
#> also installing the dependencies 'Rcpp', 'utf8', 'askpass', 'credentials', 'openssl', 'sys', 'zip', 'gitcreds', 'httr2', 'ini', 'httpuv', 'xtable', 'sourcetools', 'later', 'promises', 'fansi', 'systemfonts', 'textshaping', 'pillar', 'pkgconfig', 'diffobj', 'rematch2', 'crayon', 'curl', 'gert', 'gh', 'purrr', 'rprojroot', 'whisker', 'shiny', 'prettyunits', 'downlit', 'httr', 'ragg', 'tibble', 'xml2', 'htmlwidgets', 'xopen', 'brew', 'commonmark', 'cpp11', 'brio', 'praise', 'waldo', 'usethis', 'desc', 'miniUI', 'pkgbuild', 'pkgdown', 'pkgload', 'profvis', 'rcmdcheck', 'remotes', 'roxygen2', 'rversions', 'sessioninfo', 'testthat', 'urlchecker'
#> package 'Rcpp' successfully unpacked and MD5 sums checked
#> package 'utf8' successfully unpacked and MD5 sums checked
#> package 'askpass' successfully unpacked and MD5 sums checked
#> package 'credentials' successfully unpacked and MD5 sums checked
#> package 'openssl' successfully unpacked and MD5 sums checked
#> package 'sys' successfully unpacked and MD5 sums checked
#> package 'zip' successfully unpacked and MD5 sums checked
#> package 'gitcreds' successfully unpacked and MD5 sums checked
#> package 'httr2' successfully unpacked and MD5 sums checked
#> package 'ini' successfully unpacked and MD5 sums checked
#> package 'httpuv' successfully unpacked and MD5 sums checked
#> package 'xtable' successfully unpacked and MD5 sums checked
#> package 'sourcetools' successfully unpacked and MD5 sums checked
#> package 'later' successfully unpacked and MD5 sums checked
#> package 'promises' successfully unpacked and MD5 sums checked
#> package 'fansi' successfully unpacked and MD5 sums checked
#> package 'systemfonts' successfully unpacked and MD5 sums checked
#> package 'textshaping' successfully unpacked and MD5 sums checked
#> package 'pillar' successfully unpacked and MD5 sums checked
#> package 'pkgconfig' successfully unpacked and MD5 sums checked
#> package 'diffobj' successfully unpacked and MD5 sums checked
#> package 'rematch2' successfully unpacked and MD5 sums checked
#> package 'crayon' successfully unpacked and MD5 sums checked
#> package 'curl' successfully unpacked and MD5 sums checked
#> package 'gert' successfully unpacked and MD5 sums checked
#> package 'gh' successfully unpacked and MD5 sums checked
#> package 'purrr' successfully unpacked and MD5 sums checked
#> package 'rprojroot' successfully unpacked and MD5 sums checked
#> package 'whisker' successfully unpacked and MD5 sums checked
#> package 'shiny' successfully unpacked and MD5 sums checked
#> package 'prettyunits' successfully unpacked and MD5 sums checked
#> package 'downlit' successfully unpacked and MD5 sums checked
#> package 'httr' successfully unpacked and MD5 sums checked
#> package 'ragg' successfully unpacked and MD5 sums checked
#> package 'tibble' successfully unpacked and MD5 sums checked
#> package 'xml2' successfully unpacked and MD5 sums checked
#> package 'htmlwidgets' successfully unpacked and MD5 sums checked
#> package 'xopen' successfully unpacked and MD5 sums checked
#> package 'brew' successfully unpacked and MD5 sums checked
#> package 'commonmark' successfully unpacked and MD5 sums checked
#> package 'cpp11' successfully unpacked and MD5 sums checked
#> package 'brio' successfully unpacked and MD5 sums checked
#> package 'praise' successfully unpacked and MD5 sums checked
#> package 'waldo' successfully unpacked and MD5 sums checked
#> package 'usethis' successfully unpacked and MD5 sums checked
#> package 'desc' successfully unpacked and MD5 sums checked
#> package 'miniUI' successfully unpacked and MD5 sums checked
#> package 'pkgbuild' successfully unpacked and MD5 sums checked
#> package 'pkgdown' successfully unpacked and MD5 sums checked
#> package 'pkgload' successfully unpacked and MD5 sums checked
#> package 'profvis' successfully unpacked and MD5 sums checked
#> package 'rcmdcheck' successfully unpacked and MD5 sums checked
#> package 'remotes' successfully unpacked and MD5 sums checked
#> package 'roxygen2' successfully unpacked and MD5 sums checked
#> package 'rversions' successfully unpacked and MD5 sums checked
#> package 'sessioninfo' successfully unpacked and MD5 sums checked
#> package 'testthat' successfully unpacked and MD5 sums checked
#> package 'urlchecker' successfully unpacked and MD5 sums checked
#> package 'devtools' successfully unpacked and MD5 sums checked
#> 
#> The downloaded binary packages are in
#>  C:\Users\Felix\AppData\Local\Temp\RtmpQD2Tr4\downloaded_packages

install.packages('https://github.com/Open-Systems-Pharmacology/rClr/releases/download/v0.9.2/rClr_0.9.2.zip', repos = NULL, type = 'binary',quiet = T)

devtools::install_github("esqLABS/esqlabsR@7f676b5", dependencies = TRUE,quiet = T)
#> Warning in untar2(tarfile, files, list, exdir, restore_times): skipping pax
#> global extended headers

#> Warning in untar2(tarfile, files, list, exdir, restore_times): skipping pax
#> global extended headers

#> Warning in untar2(tarfile, files, list, exdir, restore_times): skipping pax
#> global extended headers

#> Warning in untar2(tarfile, files, list, exdir, restore_times): skipping pax
#> global extended headers

#> Warning in untar2(tarfile, files, list, exdir, restore_times): skipping pax
#> global extended headers

#> Warning in untar2(tarfile, files, list, exdir, restore_times): skipping pax
#> global extended headers
#> Installing 11 packages: colorspace, viridisLite, RColorBrewer, munsell, labeling, farver, scales, isoband, gtable, ggplot2, patchwork
#> package 'colorspace' successfully unpacked and MD5 sums checked
#> package 'viridisLite' successfully unpacked and MD5 sums checked
#> package 'RColorBrewer' successfully unpacked and MD5 sums checked
#> package 'munsell' successfully unpacked and MD5 sums checked
#> package 'labeling' successfully unpacked and MD5 sums checked
#> package 'farver' successfully unpacked and MD5 sums checked
#> package 'scales' successfully unpacked and MD5 sums checked
#> package 'isoband' successfully unpacked and MD5 sums checked
#> package 'gtable' successfully unpacked and MD5 sums checked
#> package 'ggplot2' successfully unpacked and MD5 sums checked
#> package 'patchwork' successfully unpacked and MD5 sums checked
#> Warning in untar2(tarfile, files, list, exdir, restore_times): skipping pax
#> global extended headers

#> Warning in untar2(tarfile, files, list, exdir, restore_times): skipping pax
#> global extended headers
#> Installing 11 packages: tidyselect, generics, bit, progress, tzdb, hms, bit64, dplyr, vroom, tidyr, readr
#> 
#>   There is a binary version available but the source version is later:
#>       binary source needs_compilation
#> dplyr  1.1.1  1.1.2              TRUE
#> 
#> package 'tidyselect' successfully unpacked and MD5 sums checked
#> package 'generics' successfully unpacked and MD5 sums checked
#> package 'bit' successfully unpacked and MD5 sums checked
#> package 'progress' successfully unpacked and MD5 sums checked
#> package 'tzdb' successfully unpacked and MD5 sums checked
#> package 'hms' successfully unpacked and MD5 sums checked
#> package 'bit64' successfully unpacked and MD5 sums checked
#> package 'vroom' successfully unpacked and MD5 sums checked
#> package 'tidyr' successfully unpacked and MD5 sums checked
#> package 'readr' successfully unpacked and MD5 sums checked
#> installing the source package 'dplyr'
#> Warning in untar2(tarfile, files, list, exdir, restore_times): skipping pax
#> global extended headers

#> Warning in untar2(tarfile, files, list, exdir, restore_times): skipping pax
#> global extended headers
#> Installing 7 packages: minqa, minpack.lm, coda, rootSolve, deSolve, hash, FME
#> package 'minqa' successfully unpacked and MD5 sums checked
#> package 'minpack.lm' successfully unpacked and MD5 sums checked
#> package 'coda' successfully unpacked and MD5 sums checked
#> package 'rootSolve' successfully unpacked and MD5 sums checked
#> package 'deSolve' successfully unpacked and MD5 sums checked
#> package 'hash' successfully unpacked and MD5 sums checked
#> package 'FME' successfully unpacked and MD5 sums checked
#> Installing 34 packages: sysfonts, AsioHeaders, websocket, backports, tidyselect, generics, hms, rematch, colorspace, viridisLite, RColorBrewer, munsell, labeling, farver, showtextdb, pingr, globals, chromote, checkmate, dplyr, progress, cellranger, scales, isoband, gtable, vdiffr, showtext, shinytest2, shinyjs, openxlsx, writexl, tidyr, readxl, ggplot2
#> 
#>   There is a binary version available but the source version is later:
#>       binary source needs_compilation
#> dplyr  1.1.1  1.1.2              TRUE
#> 
#> package 'sysfonts' successfully unpacked and MD5 sums checked
#> package 'AsioHeaders' successfully unpacked and MD5 sums checked
#> package 'websocket' successfully unpacked and MD5 sums checked
#> package 'backports' successfully unpacked and MD5 sums checked
#> package 'tidyselect' successfully unpacked and MD5 sums checked
#> package 'generics' successfully unpacked and MD5 sums checked
#> package 'hms' successfully unpacked and MD5 sums checked
#> package 'rematch' successfully unpacked and MD5 sums checked
#> package 'colorspace' successfully unpacked and MD5 sums checked
#> package 'viridisLite' successfully unpacked and MD5 sums checked
#> package 'RColorBrewer' successfully unpacked and MD5 sums checked
#> package 'munsell' successfully unpacked and MD5 sums checked
#> package 'labeling' successfully unpacked and MD5 sums checked
#> package 'farver' successfully unpacked and MD5 sums checked
#> package 'showtextdb' successfully unpacked and MD5 sums checked
#> package 'pingr' successfully unpacked and MD5 sums checked
#> package 'globals' successfully unpacked and MD5 sums checked
#> package 'chromote' successfully unpacked and MD5 sums checked
#> package 'checkmate' successfully unpacked and MD5 sums checked
#> package 'progress' successfully unpacked and MD5 sums checked
#> package 'cellranger' successfully unpacked and MD5 sums checked
#> package 'scales' successfully unpacked and MD5 sums checked
#> package 'isoband' successfully unpacked and MD5 sums checked
#> package 'gtable' successfully unpacked and MD5 sums checked
#> package 'vdiffr' successfully unpacked and MD5 sums checked
#> package 'showtext' successfully unpacked and MD5 sums checked
#> package 'shinytest2' successfully unpacked and MD5 sums checked
#> package 'shinyjs' successfully unpacked and MD5 sums checked
#> package 'openxlsx' successfully unpacked and MD5 sums checked
#> package 'writexl' successfully unpacked and MD5 sums checked
#> package 'tidyr' successfully unpacked and MD5 sums checked
#> package 'readxl' successfully unpacked and MD5 sums checked
#> package 'ggplot2' successfully unpacked and MD5 sums checked
#> installing the source package 'dplyr'

library(esqlabsR)
#> Loading required package: ospsuite
#> Loading required package: rClr
#> Loading the dynamic library for Microsoft .NET runtime...
#> Loaded Common Language Runtime version 4.0.30319.42000
```

<sup>Created on 2023-04-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

File: DESCRIPTION [hunk 1]
```diff
@@ -2,50 +2,60 @@ Type: Package
...
-      person("Indrajeet", "Patil", , "patilindrajeet.science@gmail.com", role = "aut",
+Authors@R: c(
+    person("esqLABS GmbH", role = c("cph", "fnd")),
+    person("Pavel", "Balazki", , "pavel.balazki@esqlabs.com", role = c("cre", "aut"),
+           comment = "https://github.com/PavelBal"),
+    person("Johanna", "Eitel", , "https://github.com/hannaei", role = "aut"),
+    person("Indrajeet", "Patil", , "patilindrajeet.science@gmail.com", role = "aut",
            comment = c(ORCID = "0000-0003-1995-6531", Twitter = "@patilindrajeets")),
-      person("Sergei", "Vavilov", , "https://github.com/svavil", role = "aut"))
+    person("Sergei", "Vavilov", , "https://github.com/svavil", role = "aut")
```
Comment (PavelBal) [inline DESCRIPTION:12]: @Felixmil Please add yourself here, too.

**Review by PavelBal — CHANGES_REQUESTED** [review]
@Felixmil Do you think the `install-packages` script should be adjusted, too? 

https://github.com/esqLABS/esqlabsR/blob/411-use-remotes_installs/inst/extdata/install-packages.R

The idea is that it installs all required packages by calling `installPackagesLocally()` (into an `renv`) or `installPackagesLocally()` (into the global library). It installs required dependencies from CRAN, downloads osp packages and installs them, etc. Can these steps be simplified with the new approach?

**Comment by Felixmil** [comment]
> The idea is that it installs all required packages by calling `installPackagesLocally()` (into an `renv`) or `installPackagesLocally()` (into the global library). It installs required dependencies from CRAN, downloads osp packages and installs them, etc. Can these steps be simplified with the new approach?

The new approach does everything automatically regarding CRAN and osp packages installations now. The remaining added value would be guidance for renv local/global installation, setup of github_pat etc...

But I'm not sure this should be an attached R script since it became way easier now. I wouldinstead write a Wiki page with copy-pastable code chunk and a link from the README (& other internal documentation).

I will write something and propose it to the team.

**Comment by PavelBal** [comment]
Then please resolve the conflict, add yourself to the authors list, and can merge.

**Review by PavelBal — APPROVED** [review]

**Pull request merged**